src/containers/Gurb/GurbValidations.js

```diff
@@ -9,18 +9,26 @@ export const identifierValidations = Yup.object().shape({
   new_contract: Yup.boolean()
     .required('CUPS_SHOULD_BE_ACTIVE')
     .oneOf([false], 'CUPS_SHOULD_BE_ACTIVE'),
+  owner: Yup.object().shape({
+    nif: Yup.string()
+      .required('ERROR_REQUIRED_FIELD')
+      .matches(/^[0-9A-Z][0-9]{7}[0-9A-Z]\d*$/, 'INVALID_NIF'),
```

Comment:
I check it out with national and nie formats and validation was successful https://regex101.com/r/35whOu/1

As I understand, we are covering: 
`X1234567A`	Letter + 7 digits + letter
`01234567Z`	Digit + 7 digits + letter
`12345678`  	Digit + 7 digits (no trailing letter, but allowed since [0-9A-Z] covers digits too)
`A12345678`	Letter + 7 digits + digit
`X1234567A123`	Extra digits at the end (allowed by \d*)